workflows/kf_single_cell_10x.cwl

```diff
@@ -34,58 +34,68 @@ requirements:
-  fastq_dir: {type: 'Directory', doc: "directory of fastqs being run"}
-  sample_name: {type: 'string[]', doc: "used as prefix for finding fastqs to analyze, e.g. 1k_PBMCs_TotalSeq_B_3p_LT_antibody if the names of the underlying fastqs are of the form 1k_PBMCs_TotalSeq_B_3p_LT_antibody_S1_L001_I1_001.fastq.gz, one per input fastq in the same order"}
-  corrected_read_1_name: {type: 'string[]?', doc: "corrected read one names in the 10x expected format 'SampleName_S1_L001_R1_001'. When provided, must be in the same order and same length as the sample name and corrected_read_2_name arrays."}
-  corrected_read_2_name: {type: 'string[]?', doc: "corrected read two names in the 10x expected format 'SampleName_S1_L001_R2_001'. When provided, must be in the same order and same length as the sample name and corrected_read_1_name arrays."}
-  reference: {type: 'Directory', doc: "directory of reference files"}
-  expected_doublet_rate: {type: 'float?', default: 0.06, doc: "expected doublet rate, usually specific to the method; default 0.06 for 10X"}
-  doublet_score_threshold: {type: 'float?', default: 0.25, doc: "doublet cut-off, cells with greater scores will be labelled as doublets; must be between 0 and 1"}
-  count_min: {type: 'int?', default: 2, doc: "minimum expression count to retain a gene"}
-  cell_min: {type: 'int?', default: 3, doc: "minimum number of cells a gene must be in to be retained"}
-  min_gene_variability_pctl: {type: 'int?', default: 85, doc: "Keep the most highly variable genes (in the top min_gene_variability_pctl percentile), as measured by the v-statistic"}
-  n_prin_comps: {type: 'int?', default: 30, doc: "Number of PCs to use for clustering"}
-  ram: {type: 'int?', default: 16, doc: "In GB"}
-  cpus: {type: 'int?', default: 1, doc: "Number of CPUs to request"}
+  # multi-step
+  output_basename: { type: string, doc: "basename used to name output files" }
+  sample_name: { type: 'string', doc: "used as prefix for finding fastqs to analyze, e.g. 1k_PBMCs_TotalSeq_B_3p_LT_antibody if the names of the underlying fastqs are of the form 1k_PBMCs_TotalSeq_B_3p_LT_antibody_S1_L001_I1_001.fastq.gz, one per input fastq in the same order" }
+  # optional concat and rename step
+  corrected_read_1_name: { type: 'string?', doc: "corrected read one names in the 10x expected format 'SampleName_S1_L001_R1_001'. When provided, must be in the same order and same length as the sample name and corrected_read_2_name arrays." }
+  corrected_read_2_name: { type: 'string?', doc: "corrected read two names in the 10x expected format 'SampleName_S1_L001_R2_001'. When provided, must be in the same order and same length as the sample name and corrected_read_1_name arrays." }
+  # cell ranger
+  cr_localcores: { type: 'int?', doc: "Num cores to use for cell ranger", default: 36 }
+  cr_instance_ram: { type: 'int?', doc: 'Ram in GB to make available to cell ranger count step', default: 64}
```

Comment:
Is cellranger that CPU heavy that it needs less then 2 gb of ram per core? Also why 36 cores instead of 32?